### DISCUSSION TITLE
Smart contracts offer pledged coins

### DISCUSSION THREAD

**Discussion by leotons** [filecoin-project/FIPs#744] [Filecoin Virtual Machine]
Can pledged coins and gas fees for sealed sectors be paid using smart contracts after the smart contract takes over the node's owner and beneficiary?

**Comment by anorth** [comment]
Gas fees must be paid by an externally-owned account actor until we get account abstraction #388. But yes pledge tokens can be deposited to the miner actor from anywhere, including smart contracts.

**Comment by leotons** [comment]
@anorth Sorry, I may not have expressed myself clearly, let me re-describe the motivation:
Our project will raise a lot of Fil, which is initially kept in the smart contract, and after reaching a certain amount, it will be used to build nodes as pledged coins sealed into sectors.
Sealing the sector requires deducting the pledged coins from the available balance, which necessitates transferring the pledged coins from the smart contract to the miner actor. This poses a problem in that if the node is not properly maintained during the sealing process, it will incur a penalty for deducting the pledged coins.
So, is there a way to deduct pledged coins and gas from the smart contract when the sector is sealed

**Reply by woshidama323** [reply]
“gas from the smart contract” is impossible as @anorth said at the moment 
and
@anorth  May I ask how to implement this？
Will appreciate it if you can give any examples or any docs for that
